resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -182,6 +183,12 @@ private[spark] class KubernetesClusterSchedulerBackend(
       .withName(env._1)
       .withValue(env._2)
       .build())
+    requiredEnv = requiredEnv ++ Seq(new EnvVarBuilder()
```

Comment:
can you keep `requiredEnv` a `val` by directly concatenating that value with the new item you're appending?  Better to keep the final reference if it's that small of a change